modules/dxp/apps/analytics/analytics-reports-web/src/test/java/com/liferay/analytics/reports/web/internal/data/provider/AnalyticsReportsDataProviderTest.java

```diff
@@ -172,6 +173,77 @@ public void testGetHistoricalReadsHistoricalMetric() throws Exception {
 		Assert.assertEquals(localDate, zonedDateTime.toLocalDate());
 	}
 
+	@Test
+	public void testGetReferringDomains() throws Exception {
+		AnalyticsReportsDataProvider analyticsReportsDataProvider =
+			new AnalyticsReportsDataProvider(
+				_getHttp(
+					Collections.singletonMap(
+						"/page-referrer-hosts",
+						JSONUtil.put(
+							"abc.com", 3.0
+						).put(
+							"google.com", 6.0
+						).put(
+							"liferay.com", 1.0
+						).toString())));
+
+		List<ReferringURL> referringURLS =
+			analyticsReportsDataProvider.getReferringDomains(
+				RandomTestUtil.randomLong(), RandomTestUtil.randomString());
+
+		Assert.assertEquals(referringURLS.toString(), 3, referringURLS.size());
+		Assert.assertEquals(
+			String.valueOf(new ReferringURL(1, "liferay.com")),
+			String.valueOf(referringURLS.get(0)));
+		Assert.assertEquals(
+			String.valueOf(new ReferringURL(6, "google.com")),
+			String.valueOf(referringURLS.get(1)));
+		Assert.assertEquals(
+			String.valueOf(new ReferringURL(3, "abc.com")),
+			String.valueOf(referringURLS.get(2)));
+	}
+
+	@Test
+	public void testGetReferringPages() throws Exception {
+		AnalyticsReportsDataProvider analyticsReportsDataProvider =
+			new AnalyticsReportsDataProvider(
+				_getHttp(
+					Collections.singletonMap(
+						"/page-referrers",
+						JSONUtil.put(
+							"https://slickdeals.net/computer-deals", 6.0
+						).put(
+							"https://slickdeals.net/credit-card-offers/", 1.0
+						).put(
+							"https://www.tomshardware.com/news/toms-hardware-" +
+								"live-events-ces-2021",
+							3.0
+						).toString())));
+
+		List<ReferringURL> referringURLS =
```

Comment:
the same here

Reply:
In Liferay we have agreed that the type of the variable should be descriptive enough, that it is the reason why we chose to use always possible the type of the variable. in the name of the variable